### PR TITLE
Revise design engine page

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | n/a
| Versions      | 3.x, master

The page was outdated as it did not cover the `base_theme` coming with Commerce.

Took the opportunity to revise it throughout.

Edit: check if the situation changes after https://github.com/ezsystems/ezcommerce-admin-ui/pull/78

### CODE REVIEW

File: docs/guide/design_engine.md [hunk 1]
```diff
@@ -1,52 +1,37 @@
...
 !!! caution
 
-    After creating a new folder either in a project's or your bundle's `templates/themes/` directory
-    you need to clear Symfony cache (`bin/console cache:clear`), even if you are working in the dev environment.
-
-- **Design**: A collection of themes.
-
-  The order of themes within a design is important as it defines the fallback order.
-  A design is identified with a name. One design can be used per SiteAccess.
+    After creating a new folder either in a project's or your bundle's `templates/themes` directory,
```
Comment (dabrt) [inline docs/guide/design_engine.md:16]:
```suggestion
    After you create a new folder either in the project's or your bundle's `templates/themes` directory,
```

File: docs/guide/design_engine.md [hunk 2]
```diff
@@ -1,52 +1,37 @@
...
+    you need to clear Symfony cache (`php bin/console cache:clear`), even if you are working in the `dev` environment.
```
Comment (dabrt) [inline docs/guide/design_engine.md:17]:
```suggestion
    clear the Symfony's cache (`php bin/console cache:clear`), even if you are working in the `dev` environment.
```

File: docs/guide/design_engine.md [hunk 3]
```diff
@@ -1,52 +1,37 @@
...
 
 ### Default designs
 
-By default two designs are included:
+By default the following designs are included:
```
Comment (dabrt) [inline docs/guide/design_engine.md:21]:
```suggestion
By default, the following designs are included:
```

File: docs/guide/design_engine.md [hunk 4]
```diff
@@ -1,52 +1,37 @@
...
 
-`admin` covers templates for the Back Office. It contains the `admin` theme.
+- `admin` covers templates for the Back Office. It contains the `admin` theme.
```
Comment (dabrt) [inline docs/guide/design_engine.md:23]:
```suggestion
- `admin` - covers templates for the Back Office, contains the `admin` theme
```

File: docs/guide/design_engine.md [hunk 5]
```diff
@@ -1,52 +1,37 @@
...
+- `standard` covers the default content rendering templates. It contains the `standard` theme.
```
Comment (dabrt) [inline docs/guide/design_engine.md:24]:
```suggestion
- `standard` - covers the default content rendering templates, contains the `standard` theme
```

File: docs/guide/design_engine.md [hunk 6]
```diff
@@ -1,52 +1,37 @@
...
+- `base_design` covers the shop templates. It contains `base_theme`.
```
Comment (dabrt) [inline docs/guide/design_engine.md:25]:
```suggestion
- `base_design` - covers the shop templates, contains `base_theme`
```

File: docs/guide/design_engine.md [hunk 7]
```diff
@@ -60,42 +45,32 @@ The standard design overrides the following Twig templates by prefixing them wit
...
+``` yaml
 ezdesign:
-    # You declare all available designs under "design_list".
     design_list:
-        # my_design will be composed of "theme1" and "theme2"
-        # "theme1" will be tried first. If a template cannot be found in "theme1", "theme2" will be tried out.
         my_design: [theme1, theme2]
+```
+
+The design engine tries `theme1` first. If it cannot find a template in `theme1`, it tries `theme2`.
```
Comment (dabrt) [inline docs/guide/design_engine.md:63]:
```suggestion
The design engine tries `theme1` first. 
If it cannot find a template in `theme1`, it tries `theme2`.
```

File: docs/guide/design_engine.md [hunk 8]
```diff
@@ -104,9 +79,29 @@ To reference current design in your code, use the `ConfigResolver` service (`ezp
 $currentDesign = $this->getConfigResolver->getParameter('design');
 ```
 
-## Design usage with assets
+## Using designs
+
+By convention, a theme directory is located under `templates/themes/<theme_name>`.
 
-For assets, a special **`ezdesign` asset package** is available.
+To indicate a template from a theme, use the `@ezdesign` special Twig namespace.
```
Comment (dabrt) [inline docs/guide/design_engine.md:86]:
```suggestion
To indicate a template from a theme, use the `@ezdesign` Twig namespace.
```

File: docs/guide/design_engine.md [hunk 9]
```diff
@@ -116,132 +111,85 @@ For assets, a special **`ezdesign` asset package** is available.
...
 
-```html+twig
-{# Will load 'some_template.html.twig' directly under one of the specified theme directories #}
-{{ include("@ezdesign/some_template.html.twig") }}
+You can also add any Twig template folder to the theme configuration.
 
-{# Will load 'another_template.html.twig', located under 'full/' directory, which is located under one of the specified theme directories #}
-{{ include("@ezdesign/full/another_template.html.twig") }}
-```
+You can use it if you want to define templates from third-party bundles as part of one of your theme,
```
Comment (dabrt) [inline docs/guide/design_engine.md:148]:
```suggestion
You can use it if you want to define templates from third-party bundles as part of one of your themes,
```

File: docs/guide/design_engine.md [hunk 10]
```diff
@@ -116,132 +111,85 @@ For assets, a special **`ezdesign` asset package** is available.
...
+or when upgrading your application in order to use the design engine,
```
Comment (dabrt) [inline docs/guide/design_engine.md:149]:
```suggestion
or when upgrading your application to use the design engine,
```

File: docs/guide/design_engine.md [hunk 11]
```diff
@@ -116,132 +111,85 @@ For assets, a special **`ezdesign` asset package** is available.
...
+if your existing templates are not yet following the convention.
```
Comment (dabrt) [inline docs/guide/design_engine.md:150]: if -> when ?

File: docs/guide/design_engine.md [hunk 12]
```diff
@@ -116,132 +111,85 @@ For assets, a special **`ezdesign` asset package** is available.
...
+        my_design: [my_theme]
+    templates_theme_paths:
+        my_theme:
+            - '%kernel.project_dir%/vendor/<vendor_name>/<bundle_name>/Resources/views'
 ```
 
-!!! tip
-
-    You may also use this notation in controllers.
+Directories following the convention always have precedence over the ones defined in `templates_theme_paths`.
```
Comment (dabrt) [inline docs/guide/design_engine.md:163]:
```suggestion
Directories that follow the convention have precedence over the ones defined in `templates_theme_paths`.
```

File: docs/guide/design_engine.md [hunk 13]
```diff
@@ -116,132 +111,85 @@ For assets, a special **`ezdesign` asset package** is available.
...
-    Directories following the convention will **always** have precedence over the ones defined
-    in config. This ensures that it is always possible to override a template from the application.
+When using themes, paths for assets are resolved at runtime.
+This is due to how the Symfony Asset component is integrated with Twig.
+This can cause significant performance impact because of I/O calls when looping over all potential theme directories,
+especially when using a lot of different designs and themes.
 
-#### Additional override paths
+To work around this issue, assets resolution can be provisioned at compilation time.
+Provisioning is the default behavior in non-debug mode (e.g. `prod` environment).
```
Comment (dabrt) [inline docs/guide/design_engine.md:182]:
```suggestion
Provisioning is the default behavior in non-debug mode (for example, in `prod` environments).
```

File: docs/guide/design_engine.md [hunk 14]
```diff
@@ -116,132 +111,85 @@ For assets, a special **`ezdesign` asset package** is available.
...
+In debug mode (e.g. `dev` environment), assets are being resolved at runtime.
```
Comment (dabrt) [inline docs/guide/design_engine.md:183]:
```suggestion
In debug mode (for example, in `dev` environments), assets are being resolved at runtime.
```

**Review by dabrt — APPROVED** [review]

File: docs/guide/design_engine.md (hunk 11 above)
Comment (DominikaK) [inline docs/guide/design_engine.md:150]:
```suggestion
and if your existing templates are not yet following the convention.
```
It's actually upgrading+existing template not following convention, so maybe this way to clarify it's not A or B or C.

**Pull request merged**